File: pf-core/src/main/java/com/purdynet/data/Downloader.java
package com.purdynet.data;

import com.purdynet.prices.PriceRecord;

import java.io.File;
import java.util.Date;
import java.util.List;

/**
 * Created with IntelliJ IDEA.
 * User: dnpurdy
 * Date: 8/17/13
 * Time: 11:37 AM
 * To change this template use File | Settings | File Templates.
 */
public interface Downloader {
    File download(String symbol, Date start, Date end);

    List<PriceRecord> getPrices(String symbol, Date start, Date end);
}

File: pf-core/src/main/scala/com/purdynet/data/Downloader.scala
package com.purdynet.data

import java.io.File
import java.util
import java.util.Date

import com.purdynet.prices.PriceRecord

trait Downloader {
  def download(symbol: String, start: Date, end: Date): File
  def getPrices(symbol: String, start: Date, end: Date): util.List[PriceRecord]
}
